pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
On Thu, Aug 4, 2016 at 5:18 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/docker/docker.go
> <https://github.com/openshift/source-to-image/pull/554#discussion_r73604610>
> :
>
> > @@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
> >  	}
> >
> >  	image, err := d.CheckImage(name)
> > -	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
> > +	if err != nil && dockerapi.IsErrImageNotFound(err) {
>
> should you be using the kube api's IsErrImageNotFound, not the docker api
> one? is that why you're seeing a difference in the error content?
>
this struct is defined in kube_docker_client.go:

```
// imageNotFoundError is the error returned by InspectImage when image not
found.
type imageNotFoundError struct {
    ID string
}
```
but there is no public method that interrogates it like what exists in
dockerapi (unless go-oracle and grep are deceiving me)

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/source-to-image/pull/554/files/63881032417630cb32a9bf66a2b033fee54d2aff..068c1c2bd2a589952a1bd740f976bca1c1fd8657#r73604610>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadClPcn1Ym2ZBdZ63VMf_X9s35zwlks5qclcMgaJpZM4JUVnn>
> .
>